### PR TITLE
BUG raise error for a not-implemented method

### PR DESCRIPTION
**Pull request by malickf**
Simple fix for https://github.com/statsmodels/statsmodels/issues/2547
I'm not sure it was necessary to write a test for it so I did it.

### CODE REVIEW

File: statsmodels/tsa/arima_model.py [hunk 1]
```diff
@@ -973,6 +973,10 @@ def fit(self, start_params=None, trend='c', method="css-mle",
         armafit.mle_settings = mlefit.mle_settings
         return ARMAResultsWrapper(armafit)
 
+    # base class of "from_formula" is "class Model(object)"
+    @classmethod
+    def from_formula(cls, formula, data, subset=None, drop_cols=None, *args, **kwargs):
+            raise NotImplementedError("from_formula is not supported for ARMA models.")
```
Comment (josef-pkt) [inline statsmodels/tsa/arima_model.py]: are these below 80 characters per line

File: statsmodels/tsa/tests/test_arima.py [hunk 2]
```diff
@@ -2301,6 +2301,8 @@ def test_arima_pickle():
     assert_almost_equal(res.fittedvalues, pkl_res.fittedvalues)
     assert_almost_equal(res.pvalues, pkl_res.pvalues)
 
+def test_arima_not_implemented():
+    assert_raises(NotImplementedError, ARIMA.from_formula,' WUE ~ 1 + SFO3 ',5)
```
Comment (josef-pkt) [inline statsmodels/tsa/tests/test_arima.py]: space after comma, pep-8
using valid data would be "nicer" but not worth the extra code, given the the exception is specified.

**Comment by josef-pkt** [comment]
Thanks, looks good
two pep-8 issues to be picky.

We don't have much exception testing yet, but it's better to add them. 
(The rest of statsmodels is still in need to get higher test coverage for warnings and exceptions.)

File: statsmodels/tsa/arima_model.py (hunk 1 above)
Comment (malickf) [reply statsmodels/tsa/arima_model.py]: I'm not aware of the 80 characters limitation, the prototype of the method is 83 character (as the original) and the second line has 75 characters (87 if I count white-space )

Comment (malickf) [reply statsmodels/tsa/arima_model.py]: I have read about the 80 pep8, I can refactor the error line to be <80, but should I also refactor the line of the method's prototype that is too loog (if yes, the original prototype, the derived or both)?

Comment (josef-pkt) [reply statsmodels/tsa/arima_model.py]: fix just where you change things or your additions
It's better not to add style changes for code not directly related to chanes in order to keep the noise out of PRs.
(We still need some style changes in many modules, but those are better left to separate PRs, and when they don't create too many merge conflicts with existing PRs.)

**Comment by josef-pkt** [comment]
FYI in case you haven't seen it yet https://www.python.org/dev/peps/pep-0008
We follow large or most parts of it, and some contributors use pep-8 tools to check.

**Comment by josef-pkt** [comment]
just an example for pep-8 cleaning
#3934
I started to fix some whitespace and line lengths that were bothering me while working on a PR. After a while I realized that the PR would include a large number of uninformative stylistic cleanups which makes the changeset difficult to read. I stashed the cleanup changes and put it afterwards in a new PR.
If it's just a few cleanup lines in neighboring code, then it is not much extra effort to review a PR or see what code was actually changed in substance.

**Comment by josef-pkt** [comment]
One more comment on github workflow:

If a commit closes a specific issue, then we can add "closes #xxxx" to the commit message and the issue is automatically closed after the merge. https://help.github.com/articles/closing-issues-using-keywords/
If we forget to do it, or it's only at the end clear that the issue can be closed, then we can add the "closes #2547" to the merge message during merge.

**Comment by malickf** [comment]
I have some troubles with the github workflow so sorry for all those commits....

Be sure that everything is ok before to merge. If it is not, I will just recreate a new branch from scratch. 

I need to study a bit more to understand how to keep my fork synchronized with the main repo.

**Comment by josef-pkt** [comment]
It's a bit messed up no, e.g. https://github.com/statsmodels/statsmodels/pull/4041/files

One general workflow comment: We (following scipy) don't merge master into a feature branch. We rebase feature branches as necessary to update it on master.

If you don't have experience yet with rebase, then I will fix this PR. It needs now an interactive rebase and squashing which is a bit more difficult than a plain rebase.

A plain rebase is similar to a merge i.e. after fetching statsmodels master
`>git rebase remotes/origin/master`

where instead of 'origin' you use your shortcut name, like what you used for merge 'remotes/statsmodels/master'

If there are no merge conflicts, then rebase finishes without problem and then the `push` needs to use `--force` to override the remote branch.

**Comment by josef-pkt** [comment]
> I have some troubles with the github workflow

I just want to mention here that everyone needed or needs a few months to get used to git and github. The advantage of git is that it is possible to recover from almost any mistake.

**Comment by malickf** [comment]
woh , thanks and I won't merge master in feature branch again .
I need some time to digest those informations... Yes, I have no experience with rebase so if you can fix my PR that will be great . Otherwise I can just delete this pull request and start a new one if you prefer. 
It's the first time I really take part in public repo so I need to learn about the github worklow.

( ps : that is  the reason why I wanted to start with "basic" fix )

Many thanks

**Comment by malickf** [comment]
I'm closing this pull request as I will do a cleaner one to fix the same problem.

**Pull request closed without merging**